src/GraphQL/Types/Composite/ComplexGraphType.cs

```diff
@@ -75,7 +75,8 @@ public virtual FieldType AddField(FieldType fieldType)
                 }
             }
 
-            NameValidator.ValidateName(fieldType.Name);
+            //check if fieldType.Name has never been set
+            NameValidator.ValidateNameNotNull(fieldType.Name);
```

Comment:
That is, at this point we now postpone the detection of the error?

Reply:
Correct.

Reply:
Do you think this is good?

Reply:
No, but it gets the job done.  If you'd like to incorporate a better suggestion in your subsequent PR, that would be just fine.  I do think we should enforce a valid schema regardless of whether a NameConverter is used to fix up names.  A separate option to disable this enforcement (breaking spec) would be fine.

See goal 3 of this PR:
> Validate that the updated names are valid during schema initialization (currently does not happen)

Reply:
If you want to push changes to this branch, that's fine too.  I've spent too much time on this already.  But I do want to meet the goals of this PR, regardless of other changes or enhancements.

Reply:
I think that it will be better if I make changes in separate PR after 3.1.2.